lib/spack/spack/spec.py

```diff
@@ -729,31 +729,28 @@ def _libs_default_handler(descriptor, spec, cls):
     # unlikely).
     name = 'lib' + spec.name.replace('-', '?')
 
-    if '+shared' in spec:
-        libs = find_libraries(
-            name, root=spec.prefix, shared=True, recursive=True
-        )
-    elif '~shared' in spec:
-        libs = find_libraries(
-            name, root=spec.prefix, shared=False, recursive=True
-        )
-    else:
-        # Prefer shared
-        libs = find_libraries(
-            name, root=spec.prefix, shared=True, recursive=True
-        )
-        if libs:
-            return libs
-
-        libs = find_libraries(
-            name, root=spec.prefix, shared=False, recursive=True
-        )
+    # To speedup the search for external packages configured e.g. in /usr,
+    # perform first non-recursive search in prefix.lib then in prefix.lib64 and
+    # finally search all of prefix recursively. The search stops when the first
+    # match is found.
+    prefix = spec.prefix
+    search_paths = [[prefix.lib, False], [prefix.lib64, False], [prefix, True]]
```

Comment:
Super minor note: for immutable objects you can use tuples:
```python
search_paths = [(prefix.lib, False), (prefix.lib64, False), (prefix, True)]
```
which are still iterable, but not extendable.